modules/guides/src/Controller/GuidesController.php

```diff
@@ -48,68 +76,70 @@ public static function create(ContainerInterface $container) {
    * @return array
    *   Render array.
    */
-  public function listGuides() {
-    $guides = [];
-    $guides_dir = DRUPAL_ROOT . (Settings::get('guides_dir') ?? '/guides');
-
-    foreach (array_diff(scandir($guides_dir), ['..', '.']) as $guide_dir) {
-      $dir = $guides_dir . '/' . $guide_dir;
-      if (is_dir($dir)) {
-        foreach (glob($dir . '/*.md') as $md) {
-          if (file_exists($md)) {
-            $parts = pathinfo($md);
-            $link = Url::fromRoute('guides.guide', ['filename' => $parts['filename']]);
-            $link = Link::fromTextAndUrl(str_replace('_', ' ', $guide_dir), $link)->toString();
-            $guides[] = [
-              '#markup' => $link,
-            ];
-          }
-        }
-      }
-    }
+  public function listGuides(): array {
+    // Get the links of the files.
+    $guides = $this->guidesStorage->getLinks();
 
+    // If the guides array isn't empty, return its content as item list.
     if (!empty($guides)) {
       return [
         '#theme' => 'item_list',
         '#items' => $guides,
       ];
     }
+    // Else return markup text.
     else {
       return [
-        '#markup' => $this->t('<strong>@text</strong>', ['@text' => 'No guides found.']),
+        '#type' => 'html_tag',
+        '#tag' => 'strong',
+        '#value' => $this->t('No guides found.'),
       ];
     }
   }
 
   /**
    * Page callback that renders a markdown file on the UI.
    *
+   * @param string $filename
+   *   The name of the guide file.
+   *
    * @return array
    *   Render array.
+   *
+   * @throws \Drupal\guides\Exception\ParseException
+   *   Thrown when no guide file is found.
    */
-  public function guideContent($filename) {
-    $guides_dir = Settings::get('guides_dir') ?? '/guides';
-    $target = [
-      'dir' => FALSE,
-      'file' => FALSE,
-    ];
+  public function guideContent(string $filename): array {
+    $cid = 'guides_files:' . $this->guidesStorage->getDirectory();
```

Comment:
Wherever I see this I consider this as a bug in the flow because this cache should be only handled by the GuidesStorage service and codes that use the storage methods should not know its existence at all.